src/olympia/blocklist/tests/test_admin.py

```diff
@@ -304,6 +313,14 @@ def test_add_single(self):
             f'The blocklist submission {FANCY_QUOTE_OPEN}No Sign-off: guid@; '
             f'dfd; some reason{FANCY_QUOTE_CLOSE} was added successfully.'
         ]
```

Comment:
This test is.... hard to follow, and I wrote it...  🐼 
So, things to change:
- lets not repeatedly assign `log` to a different thing.  Where we don't need to reference the log entry again just don't assign it a var.  Otherwise have sensible naming.
- assert that `len(logs) == 5` - and if it isn't, assert what the extras log entries are (so we know there aren't additional block or reject actions)
- if we're asserting that there's only 1 log for the disabled version we kinda need to assert there are the expected log entries for the other versions.  

I think the following is a better way of checking the entries as as expected (note, not tested, I'm just writing it directly in here!):

```
        logs = ActivityLog.objects.for_addons(addon)
        assert len(logs) == 5
        assert logs[0].action == amo.LOG.CHANGE_STATUS.id
        assert logs[1].action == amo.LOG.CHANGE_STATUS.id
        reject_log1 = logs[2]
        assert reject_log1.action == amo.LOG.REJECT_VERSION.id
        reject_log2 = logs[3]
        assert reject_log2.action == amo.LOG.REJECT_VERSION.id
        block_log = logs[4]
        assert block_log.action == amo.LOG.BLOCKLIST_BLOCK_ADDED.id
        assert block_log.arguments == [addon, addon.guid, block]
        assert block_log.details['min_version'] == '0'
        assert block_log.details['max_version'] == addon.current_version.version
        assert block_log.details['reason'] == 'some reason'
        assert block_log == (
            ActivityLog.objects.for_block(block).filter(action=block_log.action).first()
        )
        assert block_log == (
            ActivityLog.objects.for_guidblock('guid@').filter(action=block_log.action).first()
        )
        assert [reject_log_1, block_log] == list(ActivityLog.objects.for_versions(first_version))
        assert [reject_log_2, block_log] == list(ActivityLog.objects.for_versions(second_version))
        assert [block_log] == list(ActivityLog.objects.for_versions(deleted_addon_version))
        assert not ActivityLog.objects.for_versions(pending_version).exists()
```

Reply:
> Wouldn't it make more sense to use `ActivityLog.objects.for_versions()`, and check the activity logs for each version?

`ActivityLog.objects.for_versions()` requires a `versions` arg though.  If you mean something like `ActivityLog.objects.for_versions([first_version, second_version, deleted_addon_version, disabled_version, deleted_version)` it seems like it'd be more difficult to sort out the response to know which activity log corresponded which version?

Reply:
Yes, I just left out the argument for brevity. Since this is an isolated test and we define which versions we create and what versions apply to the block, it should be easy to write that down? I feel that would make reading and understanding the test a lot clearer.

Rather than getting the activity logs for all versions at once, we'd get the logs for each version separately and compare the length and individual items to what we'd expect based on what should have happened to the version during the block (block activity vs block activity+reject activity).